internal/rand/rand.go

```diff
@@ -2,19 +2,23 @@ package rand
 
 import (
 	"math/rand"
+	"sync"
 	"time"
 )
 
 const charset = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"
 
 var seededRand *rand.Rand = rand.New( //nolint:gosec
 	rand.NewSource(time.Now().UnixNano()))
+var seededRandLock sync.Mutex
```

Comment:
I added a Mutex here because random number sources are not concurrency safe and have the potential to panic when called in multiple goroutines. We were using rand.Read before which uses a Mutex internally.